From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Fri, 16 Feb 2024 15:34:18 -0800
Subject: [PATCH] skip level.dat writing


diff --git a/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java b/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
index 399da9d43aefbb95897df4697860d5bce5317152..5f344a920e4ab372cc2113e946b992d6585d9135 100644
--- a/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -600,6 +600,7 @@ public class LevelStorageSource {
         }
 
         public void saveDataTag(RegistryAccess registryManager, WorldData saveProperties, @Nullable CompoundTag nbt) {
+            if (true) return; // The Archive - don't save level.dat
             CompoundTag nbttagcompound1 = saveProperties.createTag(registryManager, nbt);
             CompoundTag nbttagcompound2 = new CompoundTag();
 
