From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 26 Jul 2025 18:08:59 -0700
Subject: [PATCH] `archiveDisableSaving` CLI arg


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 9d4d97fcb0f3cc94e9a263c61cc0e7e58fd322c3..cf345f8e87862c67eb4b29c3e7c7765013e0cdf1 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -174,6 +174,12 @@ public class Main {
                         .ofType(String.class)
                         .defaultsTo("Unknown Server")
                         .describedAs("Name");
+
+                this.accepts("archiveDisableSaving", "Disables world and entity saving, saving is disabled by default unless forceUpgrade is set")
+                    .withOptionalArg()
+                    .ofType(Boolean.class)
+                    .defaultsTo(true)
+                    .describedAs("Disable saving");
             }
         };
 
