From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 26 Jul 2025 18:08:59 -0700
Subject: [PATCH] `archiveDisableSaving` CLI arg


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 7675acdca2162e403f4ff523bcdd6efe0cb42d49..62993aea31eda6717cdc1c47272a926fc3d7a814 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -169,6 +169,8 @@ public class Main {
                         .ofType(String.class)
                         .defaultsTo("Unknown Server")
                         .describedAs("Name");
+
+                this.accepts("archiveDisableSaving", "Disables world and entity saving, saving is disabled by default unless forceUpgrade is set");
             }
         };
 
