From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:22:20 -0800
Subject: [PATCH] Disable Region File IO Saving


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java b/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
index aa3d5a6ea59ac77665f2cfaa56a7d383016d08d0..ccd766579800dbddccfb403ea4b52cbed6b219bf 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/io/MoonriseRegionFileIO.java
@@ -492,6 +492,7 @@ public final class MoonriseRegionFileIO {
 
     private static void scheduleSave(final ServerLevel world, final int chunkX, final int chunkZ, final Consumer<BiConsumer<CompoundTag, Throwable>> scheduler,
                                      final PrioritisedExecutor.PrioritisedTask writeTask, final RegionFileType type, final Priority priority) {
+        if (archive.ArchiveSettings.disableSaving()) return; // The Archive - don't save world data
         final RegionDataController taskController = getControllerFor(world, type);
 
         final boolean[] created = new boolean[1];
@@ -879,7 +880,7 @@ public final class MoonriseRegionFileIO {
             throw new IOException(ex);
         }
     }
-    
+
     private static final class CancellableRead implements Cancellable {
 
         private BiConsumer<CompoundTag, Throwable> callback;
