From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:41:46 -0800
Subject: [PATCH] Skip chunk save serialization


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
index 9a9a599ef178f851ee5c783631a724013a693586..edaabfd3dfcd75ce4436c6b8ea9126b6b162df7e 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
@@ -1704,6 +1704,10 @@ public final class NewChunkHolder {
     }
 
     private boolean saveChunk(final ChunkAccess chunk, final boolean unloading, final Completable<CompoundTag>[] chunkSave) {
+        // The Archive - Skip chunk serialization
+        if (archive.ArchiveSettings.disableSaving()) {
+            return false;
+        }
         if (!chunk.isUnsaved()) {
             return false;
         }
