From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:43:07 -0800
Subject: [PATCH] Skip entity save serialization


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
index edaabfd3dfcd75ce4436c6b8ea9126b6b162df7e..86fd2df9f8221c8472261c0621112d4717f4f2ed 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
@@ -1758,6 +1758,8 @@ public final class NewChunkHolder {
 
     private boolean lastEntitySaveNull;
     private boolean saveEntities(final ChunkEntitySlices entities, final boolean unloading, final Completable<CompoundTag>[] entitySave) {
+        // The Archive - skip entity save
+        if (archive.ArchiveSettings.disableSaving()) return false;
         try {
             if (entities.isTransient()) {
                 if (!unloading) {
@@ -1814,6 +1816,8 @@ public final class NewChunkHolder {
 
     private boolean lastPoiSaveNull;
     private boolean savePOI(final PoiChunk poi, final boolean unloading, final Completable<CompoundTag>[] poiSave) {
+        // The Archive - skip POI save
+        if (archive.ArchiveSettings.disableSaving()) return false;
         try {
             final CompoundTag save = poi.save();
             poi.setDirty(false);
