From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:43:07 -0800
Subject: [PATCH] Skip entity save serialization


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
index d2a737008b4f0c1a04e92d27fe4e1ff50f5f6cdc..133cc947ec493bb0b1a6cb3072b70221103358de 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
@@ -1796,6 +1796,8 @@ public final class NewChunkHolder {
     private boolean lastEntitySaveNull;
     private CompoundTag lastEntityUnload;
     private boolean saveEntities(final ChunkEntitySlices entities, final boolean unloading) {
+        // The Archive - skip entity save
+        if (archive.ArchiveSettings.disableSaving()) return false;
         try {
             CompoundTag mergeFrom = null;
             if (entities.isTransient()) {
@@ -1838,6 +1840,8 @@ public final class NewChunkHolder {
 
     private boolean lastPoiSaveNull;
     private boolean savePOI(final PoiChunk poi, final boolean unloading) {
+        // The Archive - skip POI save
+        if (archive.ArchiveSettings.disableSaving()) return false;
         try {
             final CompoundTag save = poi.save();
             poi.setDirty(false);
