From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:43:07 -0800
Subject: [PATCH] Skip entity save serialization


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
index 7d4b44c38065faf51fbaace09536fed8d5d5ced8..619d5f283a6fbebafed2d9195f1a2cb064976fe1 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/scheduling/NewChunkHolder.java
@@ -1803,7 +1803,8 @@ public final class NewChunkHolder {
                     return false;
                 }
                 try {
-                    mergeFrom = MoonriseRegionFileIO.loadData(this.world, this.chunkX, this.chunkZ, MoonriseRegionFileIO.RegionFileType.ENTITY_DATA, Priority.BLOCKING);
+                    // The Archive - skip entity save serialization
+                    // mergeFrom = MoonriseRegionFileIO.loadData(this.world, this.chunkX, this.chunkZ, MoonriseRegionFileIO.RegionFileType.ENTITY_DATA, Priority.BLOCKING);
                 } catch (final Exception ex) {
                     LOGGER.error("Cannot merge transient entities for chunk (" + this.chunkX + "," + this.chunkZ + ") in world '" + WorldUtil.getWorldName(this.world) + "', data on disk will be replaced", ex);
                 }
@@ -1815,14 +1816,16 @@ public final class NewChunkHolder {
                     // don't override the data on disk with nothing
                     return false;
                 } else {
-                    ChunkEntitySlices.copyEntities(mergeFrom, save);
+                    // The Archive - skip entity save serialization
+                    // ChunkEntitySlices.copyEntities(mergeFrom, save);
                 }
             }
             if (save == null && this.lastEntitySaveNull) {
                 return false;
             }
 
-            MoonriseRegionFileIO.scheduleSave(this.world, this.chunkX, this.chunkZ, save, MoonriseRegionFileIO.RegionFileType.ENTITY_DATA);
+            // The Archive - skip entity save serialization
+            // MoonriseRegionFileIO.scheduleSave(this.world, this.chunkX, this.chunkZ, save, MoonriseRegionFileIO.RegionFileType.ENTITY_DATA);
             this.lastEntitySaveNull = save == null;
             if (unloading) {
                 this.lastEntityUnload = save;
