From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:44:49 -0800
Subject: [PATCH] Skip level.dat writing


diff --git a/net/minecraft/world/level/storage/LevelStorageSource.java b/net/minecraft/world/level/storage/LevelStorageSource.java
index 8db7167fab968c9e99211ce95e5d503f9d2eb5f0..ab069be9fa50002db7c67f4ce0f881c7f1147087 100644
--- a/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -576,6 +576,8 @@ public class LevelStorageSource {
         }
 
         public void saveDataTag(final WorldData levelData, final @Nullable UUID singleplayerUUID) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             CompoundTag dataTag = levelData.createTag(singleplayerUUID);
             CompoundTag root = new CompoundTag();
             root.put("Data", dataTag);
@@ -590,6 +592,8 @@ public class LevelStorageSource {
         }
 
         private void saveLevelData(final CompoundTag root) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             Path worldDir = this.levelDirectory.path();
 
             try {
