From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:44:49 -0800
Subject: [PATCH] Skip level.dat writing


diff --git a/net/minecraft/world/level/storage/LevelStorageSource.java b/net/minecraft/world/level/storage/LevelStorageSource.java
index 3b40822ea6ec9783fe3cb8eaba069a8d626d8382..8766fa81a71664caee903b1d9b3140de5a44217e 100644
--- a/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -512,6 +512,8 @@ public class LevelStorageSource {
         }
 
         public void saveDataTag(RegistryAccess registries, WorldData serverConfiguration, @Nullable CompoundTag hostPlayerNBT) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             CompoundTag compoundTag = serverConfiguration.createTag(registries, hostPlayerNBT);
             CompoundTag compoundTag1 = new CompoundTag();
             compoundTag1.put("Data", compoundTag);
@@ -519,6 +521,8 @@ public class LevelStorageSource {
         }
 
         private void saveLevelData(CompoundTag tag) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             Path path = this.levelDirectory.path();
 
             try {
