From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:44:49 -0800
Subject: [PATCH] Skip level.dat writing


diff --git a/net/minecraft/world/level/storage/LevelStorageSource.java b/net/minecraft/world/level/storage/LevelStorageSource.java
index 6f24eb34e0ecc2b03371c585a9538e9337a2295a..14ef96af1757e5aa31b4af38d5c05950382abf40 100644
--- a/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -583,6 +583,8 @@ public class LevelStorageSource {
         }
 
         public void saveDataTag(final WorldData levelData, final @Nullable UUID singleplayerUUID) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             CompoundTag dataTag = levelData.createTag(singleplayerUUID);
             CompoundTag root = new CompoundTag();
             root.put("Data", dataTag);
@@ -597,6 +599,8 @@ public class LevelStorageSource {
         }
 
         private void saveLevelData(final CompoundTag root) {
+            // The Archive - don't save level.dat
+            if (archive.ArchiveSettings.disableSaving()) return;
             Path worldDir = this.levelDirectory.path();
 
             try {
