From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:46:19 -0800
Subject: [PATCH] Skip RegionFileStorage write


diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index 8ea830f561f5b7d6ff6ca3c38adc7fe8e54e4cc7..17d503c77d5e1205d40ba2fffcf0e1e5f52a8ea4 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -411,6 +411,8 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
             if (chunkData == null) {
                 regionFile.clear(chunkPos);
             } else {
+            // The Archive - skip chunk writing
+            if (archive.ArchiveSettings.disableSaving()) return;
             DataOutputStream chunkDataOutputStream = regionFile.getChunkDataOutputStream(chunkPos); // Paper - Only write if successful
             try { // Paper - Only write if successful
                     NbtIo.write(chunkData, chunkDataOutputStream);
