From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:46:19 -0800
Subject: [PATCH] Skip RegionFileStorage write


diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index dad7f94b611cf0fc68b1a3878c458233f6bb6d61..3ee42e2c15ae62748db6f180fb08349a53f7c0f8 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -384,6 +384,8 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         if (chunkData == null) {
             regionFile.clear(chunkPos);
         } else {
+            // The Archive - skip chunk writing
+            if (true) return;
             DataOutputStream chunkDataOutputStream = regionFile.getChunkDataOutputStream(chunkPos); // Paper - Only write if successful
             try { // Paper - Only write if successful
                 NbtIo.write(chunkData, chunkDataOutputStream);
