From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 15 Feb 2025 23:46:19 -0800
Subject: [PATCH] Skip RegionFileStorage write


diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index 63b40c420030d935bb81a219fb33defe0946e21f..19149a6d5cb03152daa28161f6aa2a5b8c34f8ef 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -397,6 +397,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
     }
 
     public void write(final ChunkPos pos, final @Nullable CompoundTag value) throws IOException {
+        if (value != null && archive.ArchiveSettings.disableSaving()) return; // The Archive - skip chunk writing
         if (!SharedConstants.DEBUG_DONT_SAVE_WORLD) {
             RegionFile region = this.getRegionFile(pos, value == null); // CraftBukkit // Paper - rewrite chunk system
             // Paper start - rewrite chunk system
