From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:06:20 -0700
Subject: [PATCH] Allow loading root entities from outside the serialized chunk


diff --git a/ca/spottedleaf/moonrise/patches/chunk_system/level/entity/EntityLookup.java b/ca/spottedleaf/moonrise/patches/chunk_system/level/entity/EntityLookup.java
index 328784ba08236553e49d62320b79f7222292412f..84cb47db7bd1df17306bff5d31e115a8b972e6d2 100644
--- a/ca/spottedleaf/moonrise/patches/chunk_system/level/entity/EntityLookup.java
+++ b/ca/spottedleaf/moonrise/patches/chunk_system/level/entity/EntityLookup.java
@@ -368,13 +368,6 @@ public abstract class EntityLookup implements LevelEntityGetter<Entity> {
                 continue;
             }
 
-            if (forChunk != null && !entity.chunkPosition().equals(forChunk)) {
-                LOGGER.warn("Root entity " + entity + " is outside of serialized chunk " + forChunk);
-                // can't set removed here, as we may not own the chunk position
-                // skip the entity
-                continue;
-            }
-
             final Vec3 rootPosition = entity.position();
 
             // always adjust positions before adding passengers in case plugins access the entity, and so that
