From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:07:41 -0700
Subject: [PATCH] Disable all data directory saving


diff --git a/net/minecraft/world/level/storage/SavedDataStorage.java b/net/minecraft/world/level/storage/SavedDataStorage.java
index 971c3f3311f5eb132817589ee5b7f65736c25b76..1f37f82be858e4b91e9c940d2fe9a88f98521c6a 100644
--- a/net/minecraft/world/level/storage/SavedDataStorage.java
+++ b/net/minecraft/world/level/storage/SavedDataStorage.java
@@ -147,6 +147,8 @@ public class SavedDataStorage implements AutoCloseable {
     }
 
     public CompletableFuture<?> scheduleSave() {
+        // The Archive - don't save data
+        if (archive.ArchiveSettings.disableSaving()) return CompletableFuture.completedFuture(null);
         if (this.closed) {
             throw new IllegalStateException("Trying to schedule save when SavedDataStorage is already closed");
         } else {
