From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:07:41 -0700
Subject: [PATCH] Disable all data directory saving


diff --git a/net/minecraft/world/level/storage/SavedDataStorage.java b/net/minecraft/world/level/storage/SavedDataStorage.java
index f359e53024019f35aaa5cd8477da708bda2fc57a..3f82661f18b5543be2fcc013e7356f397a7f590a 100644
--- a/net/minecraft/world/level/storage/SavedDataStorage.java
+++ b/net/minecraft/world/level/storage/SavedDataStorage.java
@@ -144,6 +144,8 @@ public class SavedDataStorage implements AutoCloseable {
     }
 
     public CompletableFuture<?> scheduleSave() {
+        // The Archive - don't save data
+        if (archive.ArchiveSettings.disableSaving()) return CompletableFuture.completedFuture(null);
         if (this.closed) {
             throw new IllegalStateException("Trying to schedule save when SavedDataStorage is already closed");
         }
