From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:07:41 -0700
Subject: [PATCH] Disable all data directory saving


diff --git a/net/minecraft/world/level/storage/DimensionDataStorage.java b/net/minecraft/world/level/storage/DimensionDataStorage.java
index 1ff0b19a077333ad9da9cab7f5a891eeaa2109a9..ed201291d0207b32c1766910488c3dd35c8e8edf 100644
--- a/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -144,6 +144,8 @@ public class DimensionDataStorage implements AutoCloseable {
     }
 
     public CompletableFuture<?> scheduleSave() {
+        // The Archive - don't save data
+        if (archive.ArchiveSettings.disableSaving()) return CompletableFuture.completedFuture(null);
         Map<SavedDataType<?>, CompoundTag> map = this.collectDirtyTagsToSave();
         if (map.isEmpty()) {
             return CompletableFuture.completedFuture(null);
