From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sun, 16 Feb 2025 00:10:02 -0800
Subject: [PATCH] Search for exit portals using chunk data


diff --git a/net/minecraft/world/level/block/NetherPortalBlock.java b/net/minecraft/world/level/block/NetherPortalBlock.java
index d3f0a61c0c91d3a6a1dc4e59d57743ba89a9c19a..5be6ac18f1073a728da4eac95be1d2aac4b3f109 100644
--- a/net/minecraft/world/level/block/NetherPortalBlock.java
+++ b/net/minecraft/world/level/block/NetherPortalBlock.java
@@ -179,7 +179,8 @@ public class NetherPortalBlock extends Block implements Portal {
 
     @Nullable
     private TeleportTransition getExitPortal(ServerLevel level, Entity entity, BlockPos pos, BlockPos exitPos, WorldBorder worldBorder, org.bukkit.craftbukkit.event.PortalEventResult result) { // CraftBukkit
-        Optional<BlockPos> optional = level.getPortalForcer().findClosestPortalPosition(exitPos, worldBorder, result.searchRadius()); // CraftBukkit
+        // The Archive - Search for exit portals from chunk data instead of POI
+        Optional<BlockPos> optional = archive.PortalScanner.findNearestPortal(level, exitPos, result.searchRadius());
         BlockUtil.FoundRectangle largestRectangleAround;
         TeleportTransition.PostTeleportTransition postTeleportTransition;
         if (optional.isPresent()) {
