From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sun, 16 Feb 2025 00:10:02 -0800
Subject: [PATCH] Search for exit portals using chunk data


diff --git a/net/minecraft/world/level/block/NetherPortalBlock.java b/net/minecraft/world/level/block/NetherPortalBlock.java
index 653b72855197733b33e310b0ccd4fbf04e22e95f..7970c425d9ae05958d3dbcc4bebdf4759f1c3dcb 100644
--- a/net/minecraft/world/level/block/NetherPortalBlock.java
+++ b/net/minecraft/world/level/block/NetherPortalBlock.java
@@ -176,7 +176,9 @@ public class NetherPortalBlock extends Block implements Portal {
     private @Nullable TeleportTransition getExitPortal(
         ServerLevel level, Entity entity, BlockPos pos, BlockPos exitPos, WorldBorder worldBorder, org.bukkit.craftbukkit.event.PortalEventResult result // CraftBukkit
     ) {
-        Optional<BlockPos> optional = level.getPortalForcer().findClosestPortalPosition(exitPos, worldBorder, result.searchRadius()); // CraftBukkit
+        // The Archive - Search for exit portals from chunk data instead of POI
+        Optional<BlockPos> optional = archive.PortalScanner.findNearestPortal(level, exitPos, result.searchRadius());
+        // CraftBukkit
         BlockUtil.FoundRectangle largestRectangleAround;
         TeleportTransition.PostTeleportTransition postTeleportTransition;
         if (optional.isPresent()) {
