From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sun, 16 Feb 2025 00:11:22 -0800
Subject: [PATCH] mute setBlockEntity logspam


diff --git a/net/minecraft/world/level/chunk/LevelChunk.java b/net/minecraft/world/level/chunk/LevelChunk.java
index de59760d27280555a334bda4f436164568cffbd6..8c4c15d1a1239eaa26afb3ff433f5c3b4e11ba9c 100644
--- a/net/minecraft/world/level/chunk/LevelChunk.java
+++ b/net/minecraft/world/level/chunk/LevelChunk.java
@@ -537,6 +537,7 @@ public class LevelChunk extends ChunkAccess implements DebugValueSource, ca.spot
         BlockPos blockPos = blockEntity.getBlockPos();
         BlockState blockState = this.getBlockState(blockPos);
         if (!blockState.hasBlockEntity()) {
+            if (true) return; // Archive - mute logspam
             // Paper start - ServerExceptionEvent
             com.destroystokyo.paper.exception.ServerInternalException e = new com.destroystokyo.paper.exception.ServerInternalException(
                 "Trying to set block entity %s at position %s, but state %s does not allow it".formatted(blockEntity, blockPos, blockState)
@@ -548,12 +549,12 @@ public class LevelChunk extends ChunkAccess implements DebugValueSource, ca.spot
             BlockState blockState1 = blockEntity.getBlockState();
             if (blockState != blockState1) {
                 if (!blockEntity.getType().isValid(blockState)) {
-                    LOGGER.warn("Trying to set block entity {} at position {}, but state {} does not allow it", blockEntity, blockPos, blockState);
+                    LOGGER.debug("Trying to set block entity {} at position {}, but state {} does not allow it", blockEntity, blockPos, blockState);
                     return;
                 }
 
                 if (blockState.getBlock() != blockState1.getBlock()) {
-                    LOGGER.warn("Block state mismatch on block entity {} in position {}, {} != {}, updating", blockEntity, blockPos, blockState, blockState1);
+                    LOGGER.debug("Block state mismatch on block entity {} in position {}, {} != {}, updating", blockEntity, blockPos, blockState, blockState1);
                 }
 
                 blockEntity.setBlockState(blockState);
