From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:14:27 -0700
Subject: [PATCH] Handle entities with invalid position, rotation, or motion


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 70b6ea8ab35e88989b5b1f5ffd64490a9d743b56..730d0cdbb8e065e25f5dfde1719eea411c01179f 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -144,6 +144,7 @@ import net.minecraft.world.scores.PlayerTeam;
 import net.minecraft.world.scores.ScoreHolder;
 import net.minecraft.world.scores.Team;
 import net.minecraft.world.waypoints.WaypointTransmitter;
+import org.bukkit.event.entity.EntityRemoveEvent;
 import org.jetbrains.annotations.Contract;
 import org.slf4j.Logger;
 
@@ -2596,8 +2597,12 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
     public void load(ValueInput input) {
         try {
             Vec3 vec3 = input.read("Pos", Vec3.CODEC).orElse(Vec3.ZERO);
+            if (archive.NanHelper.isInvalid(vec3.x(), vec3.y(), vec3.z())) {
+                this.setRemoved(RemovalReason.DISCARDED, EntityRemoveEvent.Cause.DESPAWN);
+                return;
+            }
             Vec3 vec31 = input.read("Motion", Vec3.CODEC).orElse(Vec3.ZERO); vec31 = io.papermc.paper.util.MCUtil.sanitizeNanInf(vec31, 0D); // Paper - avoid setting NaN values
-            Vec2 vec2 = input.read("Rotation", Vec2.CODEC).orElse(Vec2.ZERO);
+            Vec2 vec2 = input.read("Rotation", Vec2.CODEC).orElse(Vec2.ZERO); vec2 = new Vec2(archive.NanHelper.ensureNonNaN(vec2.x), archive.NanHelper.ensureNonNaN(vec2.y));
             this.setDeltaMovement(Math.abs(vec31.x) > 10.0 ? 0.0 : vec31.x, Math.abs(vec31.y) > 10.0 ? 0.0 : vec31.y, Math.abs(vec31.z) > 10.0 ? 0.0 : vec31.z);
             this.hasImpulse = true;
             double d = 3.0000512E7;
