From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Wed, 23 Jul 2025 16:14:27 -0700
Subject: [PATCH] Handle entities with invalid position, rotation, or motion


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 82f6582a92f17bdeb3218c4c1fcac86981fd9c37..2a683852a8bf586c5c8e9ece26707c033ebee709 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -2689,8 +2689,12 @@ public abstract class Entity
     public void load(final ValueInput input) {
         try {
             Vec3 pos = input.read("Pos", Vec3.CODEC).orElse(Vec3.ZERO);
-            Vec3 motion = input.read("Motion", Vec3.CODEC).orElse(Vec3.ZERO);
-            Vec2 rotation = input.read("Rotation", Vec2.CODEC).orElse(Vec2.ZERO);
+            if (archive.NanHelper.isInvalid(pos.x(), pos.y(), pos.z())) {
+                this.setRemoved(RemovalReason.DISCARDED, org.bukkit.event.entity.EntityRemoveEvent.Cause.DESPAWN);
+                return;
+            }
+            Vec3 motion = input.read("Motion", Vec3.CODEC).orElse(Vec3.ZERO); motion = new Vec3(archive.NanHelper.ensureNonNaN(motion.x), archive.NanHelper.ensureNonNaN(motion.y), archive.NanHelper.ensureNonNaN(motion.z));
+            Vec2 rotation = input.read("Rotation", Vec2.CODEC).orElse(Vec2.ZERO); rotation = new Vec2(archive.NanHelper.ensureNonNaN(rotation.x), archive.NanHelper.ensureNonNaN(rotation.y));
             this.setDeltaMovement(
                 Math.abs(motion.x) > 10.0 ? 0.0 : motion.x, Math.abs(motion.y) > 10.0 ? 0.0 : motion.y, Math.abs(motion.z) > 10.0 ? 0.0 : motion.z
             );
