From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 26 Jul 2025 19:00:56 -0700
Subject: [PATCH] log archive saving state on startup


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index 6a5d404e570ede5d49c82bccc30aa75ddd58983b..da88f9e2c5f96885325d0725524de98b09434832 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -358,6 +358,8 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
             LOGGER.warn("To change this, set \"online-mode\" to \"true\" in the server.properties file.");
         }
 
+        LOGGER.info("[The Archive] World data saving is {}.", archive.ArchiveSettings.disableSaving() ? "disabled" : "enabled");
+
         // CraftBukkit start
         /*
         if (this.convertOldUsers()) {
