From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Sat, 26 Jul 2025 18:40:25 -0700
Subject: [PATCH] Disable region file padding saving


diff --git a/net/minecraft/world/level/chunk/storage/RegionFile.java b/net/minecraft/world/level/chunk/storage/RegionFile.java
index e077797cefffcb9db627cab12c7d88245e5baa69..b7920837dd5663bed3b0999a359e25d5c835d320 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFile.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFile.java
@@ -882,6 +882,8 @@ public class RegionFile implements AutoCloseable, ca.spottedleaf.moonrise.patche
     }
 
     private void padToFullSector() throws IOException {
+        // The Archive - disable writing padding if saving is disabled
+        if (archive.ArchiveSettings.disableSaving()) return;
         int fileSize = (int)this.file.size();
         int paddedSize = sizeToSectors(fileSize) * 4096;
         if (fileSize != paddedSize) {
