From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Thu, 15 Feb 2024 14:57:33 -0800
Subject: [PATCH] Disable saving


diff --git a/src/main/java/io/papermc/paper/chunk/system/io/RegionFileIOThread.java b/src/main/java/io/papermc/paper/chunk/system/io/RegionFileIOThread.java
index 2934f0cf0ef09c84739312b00186c2ef0019a165..dbed34b2d8c28a29b6dddf236f67d55af09cfc32 100644
--- a/src/main/java/io/papermc/paper/chunk/system/io/RegionFileIOThread.java
+++ b/src/main/java/io/papermc/paper/chunk/system/io/RegionFileIOThread.java
@@ -548,6 +548,7 @@ public final class RegionFileIOThread extends PrioritisedQueueExecutorThread {
 
     void scheduleSaveInternal(final ServerLevel world, final int chunkX, final int chunkZ, final CompoundTag data,
                               final RegionFileType type, final PrioritisedExecutor.Priority priority) {
+        if (true) return; // The Archive - don't save world data
         final ChunkDataController taskController = this.getControllerFor(world, type);
 
         final boolean[] created = new boolean[1];
