From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Fri, 23 Aug 2024 12:43:21 -0700
Subject: [PATCH] skip level.dat writing


diff --git a/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java b/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
index 85ba843ce7e1f62971e736fa2cc028c47b274ce4..e77d132aeadfeca5c6581e51797075b320e7c289 100644
--- a/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
+++ b/src/main/java/net/minecraft/world/level/storage/LevelStorageSource.java
@@ -601,6 +601,7 @@ public class LevelStorageSource {
         }
 
         public void saveDataTag(RegistryAccess registryManager, WorldData saveProperties, @Nullable CompoundTag nbt) {
+            if (true) return; // The Archive - don't save level.dat
             CompoundTag nbttagcompound1 = saveProperties.createTag(registryManager, nbt);
             CompoundTag nbttagcompound2 = new CompoundTag();
 
