From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Fri, 23 Aug 2024 12:44:27 -0700
Subject: [PATCH] Skip RegionFileStorage write


diff --git a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index 40689256711cc94a806ca1da346f4f62eda31526..e0ccde319205dc04ae7d6a264d4ec71b9003f7b8 100644
--- a/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/src/main/java/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -325,6 +325,8 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
         if (nbt == null) {
             regionfile.clear(pos);
         } else {
+            // The Archive - skip chunk writing
+            if (true) return;
             DataOutputStream dataoutputstream = regionfile.getChunkDataOutputStream(pos);
 
             try {
