From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: rfresh2 <89827146+rfresh2@users.noreply.github.com>
Date: Fri, 23 Aug 2024 12:49:37 -0700
Subject: [PATCH] Disable all data directory saving


diff --git a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
index 6e23e69abd56eeda3b52a22019e1b74ae10682e7..882ff84006a4450b2a451bc369d3086d6298abd6 100644
--- a/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
+++ b/src/main/java/net/minecraft/world/level/storage/DimensionDataStorage.java
@@ -134,6 +134,7 @@ public class DimensionDataStorage implements java.io.Closeable { // Paper - Writ
     // Paper end - Write SavedData IO async
 
     public void save(boolean async) { // Paper - Write SavedData IO async
+        if (true) return; // The Archive - don't save data
         this.cache.forEach((id, state) -> {
             if (state != null) {
                 // Paper start - Write SavedData IO async
